File: 144-binary-tree-preorder-traversal/binary-tree-preorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> preord(TreeNode* root,vector<int> &sol)
    {
        if(root==nullptr){
            return sol;
        }
        sol.push_back(root->val);
        preord(root->left,sol);
        preord(root->right,sol);
        return sol;
        
    }
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> sol;
        sol=preord(root,sol);
        return sol;
    }
};

File: 144-binary-tree-preorder-traversal/binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    
    def preorderTraversal(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: List[int]
        """
        res=[]
        def preorder(root):
            if not root :
                return []
            res.append(root.val)
            preorder(root.left)
            preorder(root.right)
            return res 
        return preorder(root)
